src/dlr_pipeline.cc

```diff
@@ -0,0 +1,162 @@
+#include "dlr_pipeline.h"
+
+#include <stdlib.h>
+#include <cstring>
+#include <fstream>
+#include <iterator>
+#include <numeric>
+
+using namespace dlr;
+
+
+void PipelineModel::SetupPipelineModel(std::vector<DLRModelPtr>& dlr_models) {
+  CHECK_GT(dlr_models.size(), 0) << "List of models is empty";
+  dlr_models_ = dlr_models;
+  count_ = dlr_models.size();
+  num_inputs_ = dlr_models[0]->GetNumInputs();
+  num_weights_ = dlr_models[0]->GetNumWeights();
+  num_outputs_ = dlr_models.back()->GetNumOutputs();
+  for (int i = 0; i < num_inputs_; i++) {
+    input_names_.push_back(dlr_models[0]->GetInputName(i));
+    input_types_.push_back(dlr_models[0]->GetInputType(i));
+    input_shapes_.push_back(dlr_models[0]->GetInputShape(i));
+  }
+  // Check previous model outputs and current model inputs compatibility
```

Comment:
For some relay VM models, the input might be unknown (`-1`) and the output dims could also unknown before running the model. Can you check for `-1` value in the shapes and bypass the size check in those cases?

Reply:
I added `PipelineModel::CheckModelsCompatibility()` method and use it in the constructor for initial models check and then in RUN method for output/input size/shape checks.